### ISSUE TITLE
Crash from AIOOB during whitelist processing

### ISSUE THREAD

**Issue by ghost** [InsomniaKitten/SmartHUD#6]
This is the crash log: https://pastebin.com/U324Lp3M

**Comment by ChloeDawn** [comment]
Are you able to replicate this with just my mod installed?

**Comment by ghost** [comment]
Yep, it crashes with only your mod in the mods folder

**Comment by ChloeDawn** [comment]
Did you also remove OptiFine?

**Comment by ghost** [comment]
Removed everything! Now that I launcher the game with just Optifine it crashed tho. I'll try launching with everything but Optifine

edit: nope, it crashed anyway

**Comment by adrwa** [comment]
Can you post a crash log from it loaded without optifine

**Comment by ChloeDawn** [comment]
This is an OptiFine crash, nothing to do with my mod. It has already been reported here: https://github.com/sp614x/optifine/issues/733

**Comment by ghost** [comment]
Ok apparently the game crashes when one or both the mods are present in the mods folder.
This is the crashlog you requested: https://pastebin.com/QaDQxTaC

The game crashes with your mod in the list and without optifine too tho

**Comment by ChloeDawn** [comment]
Can I see your whitelist, please?

**Comment by ghost** [comment]
I'm sorry, what do you mean with whitelist?

**Comment by ChloeDawn** [comment]
Your item whitelist, there is a config file with whitelisted items that can be shown on the HUD.

**Comment by ghost** [comment]
```
# Configuration file

whitelist {
    # Configure items that will be displayed on the HUD when present in the players inventory.
    # Follow the format modid:resourcename:metadata otherwise the item will not be registered.
    # Metadata is not required, and not defining it will default the check to any metadata.
    # This information can be obtained via Advanced Tooltips (F3+H) in-game.
    S:"Item List" <
        minecraft:clock
        minecraft:compass
        randomthings:goldencompass
        endercompass:ender_compass
        toughasnails:thermometer
        toughasnails:season_clock
        minecraft:arrow
        
     >

    # Should the HUD use the configurable whitelist when checking for valid items?
    # If false, Smart HUD will fall back to only checking for vanilla clocks and compasses.
    B:"Use Whitelist"=true
}



```

**Comment by ChloeDawn** [comment]
Fixed in [1.12-1.0.4](https://minecraft.curseforge.com/projects/smart-hud/files/2443261), hopefully.

**Comment by ghost** [comment]
Yep, it fixed it. Thanks :) 👍
